checkstyle-tester/README.md

```diff
@@ -65,6 +65,23 @@ This option is useful for Windows users where they are restricted to maximum dir
 to maven during the Checkstyle regression run.  For example, if you want to skip site generation, you
 would add `--extraMvnRegressionOptions "-Dmaven.site.skip=true"` to `diff.groovy` execution.
 
+**failsOnError** (f) - this option tells maven to fail if a checkstyle error is encountered during
+report generation. This means that your configuration uses the `<property name="severity" value="warning"/>
+` property.  If Checkstyle reports a violation for any of the projects that are being tested, the report
+generation will fail.  This option is good for usage in Checkstyle CI for no-error testing
+(optional, default is false).
+
+**allowExcludes** (g) - this option tells `diff.groovy` to allow paths and files defined in a
+`projects*.properties` file to be excluded from report generation (optional, default is false).
+
+**checkstyleVersion** (cv) - used to set the Checkstyle version to be used by maven during report
+generation; this option should mainly be used by satiellite projects
```

Comment:
Maybe I lost track. Why do we need to pass in a checkstyle version to override what is being used automatically? Point to example if there is one.

Reply:
https://github.com/sevntu-checkstyle/sevntu.checkstyle/pull/828/files#r599509434

Reply:
ok, this is just for no exception testing in sevntu.

These 2 properties are pretty dangerous to use. If the wrong version is set, maven won't fail normally. It will try to find this version installed in the repo or pulled from the server. If it finds a version to use, it will use that and not be noticeable to the user.

Take our own regression, someone active in the repo will always be running local regression. So it is possible they have previous versions installed. If the use this and set an old version, the report will be generated however it will not be hitting their new code at all and only hitting old. This will give a false sense of testing, especially if no changes are to be expected.

I would add some verbage to this to make it clear it is not to be used in normal circumstances. It is for internal use only.

Ideally, I think these should be removed but I don't see an alternative to get no exception testing to work in sevntu. If support is added for sevntu, these properties should then be removed.

Reply:
I can add a check to make sure that these are only used in single mode, too.

Reply:
```bash

➜  checkstyle-tester git:(issue-529) ✗ groovy diff.groovy -l projects-to-test-on.properties -r ../../checkstyle -p my-issue --checkstyleVersion "8.41.1"

Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=gasp -Dawt.useSystemAAFontSettings=gasp 
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/usr/share/groovy/lib/groovy-2.5.13.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Error: Diff mode may not use '--sevntuVersion' or '--checkstyleVersion' arguments!
Caught: java.lang.IllegalArgumentException: Error: invalid command line arguments!
...

➜  checkstyle-tester git:(issue-529) ✗ groovy diff.groovy -l projects-to-test-on.properties -r ../../checkstyle -p my-issue --sevntuVersion "8.41.1"

Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=gasp -Dawt.useSystemAAFontSettings=gasp 
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/usr/share/groovy/lib/groovy-2.5.13.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Error: Diff mode may not use '--sevntuVersion' or '--checkstyleVersion' arguments!
Caught: java.lang.IllegalArgumentException: Error: invalid command line arguments!
...

```

Reply:
These options have been removed now.